sdk/core/Azure.Core/src/Pipeline/Internal/RetryPolicy.cs

```diff
@@ -51,76 +41,59 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
 
                 try
                 {
-                    if (async)
-                    {
-                        await ProcessNextAsync(message, pipeline).ConfigureAwait(false);
-                    }
-                    else
-                    {
-                        ProcessNext(message, pipeline);
-                    }
+                    ProcessNext(message, pipeline);
```

Comment:
Is there a functional change here?

Reply:
No, refactoring

Reply:
Not sure I like it. The idea was to keep a single copy of the method because it's pretty complicated and differences are easy to miss.

Reply:
Sync/async mix is a bug prone as well (for example: https://github.com/Azure/azure-sdk-for-net/blob/95e76cde5f69c78d0b1fdf32346181adaa079d1b/sdk/identity/Azure.Identity/src/InteractiveBrowserCredential.cs#L102).
And while issues with missing logic is easy to find with tests that automatically run both sync and async version of the policy methods, problems with awaiters and sync contexts can appear only in user code.

I can try write an analyzer that will check if a private async method is called from sync method and then enforce it to have `isAsync` parameter and use that parameter to call sync versions of methods (when they are available). That can help with our own code, but there can be issues with third party components.